DatadogInternal/Sources/Upload/DataCompression.swift

```diff
@@ -16,7 +16,7 @@ import zlib
 /// - https://zlib.net/
 /// - https://datatracker.ietf.org/doc/html/rfc1950
 /// - https://datatracker.ietf.org/doc/html/rfc1951
-/* public */ internal struct Deflate {
+public struct Deflate {
```

Comment:
question/ Do we need compression things in public interface? The `URLRequestBuilder` handles compression:
```swift
// URLRequestBuilder.swift

func uploadRequest(with body: Data, compress: Bool = true)
```
so details of the actual `encode()`, `compress()` and `addler32()` could be internal, no?